In [114]:
from pandas import Series, DataFrame
import pandas as pd
import matplotlib.pyplot as plt
import datetime
%matplotlib inline

In [2]:
data=pd.read_csv("C:\Users\sapan_jain\Desktop\Learnings\Greedy\ggevent\ggevent.log")

In [3]:
data.columns

Index([u'{"headers": {"ai5": "8fa683e59c02c04cb781ac689686db07"',
       u' "debug": null', u' "random": null', u' "sdkv": "7.6"}',
       u' "post": {"event": "ggstart"', u' "ts": "1462759195259"}',
       u' "params": {}',
       u' "bottle": {"timestamp": "2016-05-09 02:00:00.004906"',
       u' "game_id": "55107008"}}'],
      dtype='object')

In [4]:
#length of data
len(data)

data.head()


{"headers": {"ai5": "8fa683e59c02c04cb781ac689686db07"   "debug": null  \
0  {"headers": {"ai5": "335644267c1d5f04eaea7bc6f...       "debug": null   
1  {"headers": {"ai5": "ca3071aad676bc963795a2b09...       "debug": null   
2  {"headers": {"ai5": "17412dec7d3d02c9b0b1c3d1c...       "debug": null   
3  {"headers": {"ai5": "f81167c854f1a0c86cab6188f...       "debug": null   
4  {"headers": {"ai5": "17412dec7d3d02c9b0b1c3d1c...       "debug": null   

    "random": null     "sdkv": "7.6"}   "post": {"event": "ggstart"  \
0   "random": null     "sdkv": "7.6"}   "post": {"event": "ggstart"   
1   "random": null     "sdkv": "7.6"}    "post": {"event": "ggstop"   
2   "random": null   "sdkv": "7.5.1"}    "post": {"event": "ggstop"   
3   "random": null     "sdkv": "7.6"}   "post": {"event": "ggstart"   
4   "random": null   "sdkv": "7.5.1"}   "post": {"event": "ggstart"   

    "ts": "1462759195259"}   "params": {}  \
0   "ts": "1462759189745"}   "params": {}   
1   "ts": "1462768152242"}   "params": {}   
2   "ts": "1462759193807"}   "params": {}   
3   "ts": "1462759191063"}   "params": {}   
4   "ts": "1462759194576"}   "params": {}   

   "bottle": {"timestamp": "2016-05-09 02:00:00.004906"  \
0   "bottle": {"timestamp": "2016-05-09 02:00:00....      
1   "bottle": {"timestamp": "2016-05-09 02:00:00....      
2   "bottle": {"timestamp": "2016-05-09 02:00:00....      
3   "bottle": {"timestamp": "2016-05-09 02:00:00....      
4   "bottle": {"timestamp": "2016-05-09 02:00:00....      

    "game_id": "55107008"}}  
0   "game_id": "55107008"}}  
1   "game_id": "55107008"}}  
2   "game_id": "10655437"}}  
3   "game_id": "55107008"}}  
4   "game_id": "10655437"}}

In [5]:
type(data)

pandas.core.frame.DataFrame

In [6]:
type(data.ix[3])


pandas.core.series.Series

In [7]:
for i in range(9):
    print data.values[0][i]

{"headers": {"ai5": "335644267c1d5f04eaea7bc6f51b1861"
 "debug": null
 "random": null
 "sdkv": "7.6"}
 "post": {"event": "ggstart"
 "ts": "1462759189745"}
 "params": {}
 "bottle": {"timestamp": "2016-05-09 02:00:00.033775"
 "game_id": "55107008"}}


In [8]:
data.values

array([['{"headers": {"ai5": "335644267c1d5f04eaea7bc6f51b1861"',
        ' "debug": null', ' "random": null', ..., ' "params": {}',
        ' "bottle": {"timestamp": "2016-05-09 02:00:00.033775"',
        ' "game_id": "55107008"}}'],
       ['{"headers": {"ai5": "ca3071aad676bc963795a2b09635cdf0"',
        ' "debug": null', ' "random": null', ..., ' "params": {}',
        ' "bottle": {"timestamp": "2016-05-09 02:00:00.036721"',
        ' "game_id": "55107008"}}'],
       ['{"headers": {"ai5": "17412dec7d3d02c9b0b1c3d1c3571c5c"',
        ' "debug": null', ' "random": null', ..., ' "params": {}',
        ' "bottle": {"timestamp": "2016-05-09 02:00:00.041006"',
        ' "game_id": "10655437"}}'],
       ..., 
       ['{"headers": {"ai5": "f74135393746293cbd3bf9572b601da6"',
        ' "debug": null', ' "random": null', ..., ' "params": {}',
        ' "bottle": {"timestamp": "2016-05-09 02:59:59.943395"',
        ' "game_id": "10655437"}}'],
       ['{"headers": {"ai5": "97178fceb72a45679

In [9]:
data_short=pd.read_table("C:\Users\sapan_jain\Desktop\Learnings\Greedy\ggevent\ggevent.log",sep=r'\s+', usecols=[2,11,18,20], names=["ai5","event","timestamp","gid"])

In [10]:
data_short.head(5)

ai5     event                    timestamp  \
0  8fa683e59c02c04cb781ac689686db07,  ggstart,  2016-05-09 02:00:00.004906,   
1  335644267c1d5f04eaea7bc6f51b1861,  ggstart,  2016-05-09 02:00:00.033775,   
2  ca3071aad676bc963795a2b09635cdf0,   ggstop,  2016-05-09 02:00:00.036721,   
3  17412dec7d3d02c9b0b1c3d1c3571c5c,   ggstop,  2016-05-09 02:00:00.041006,   
4  f81167c854f1a0c86cab6188f9995824,  ggstart,  2016-05-09 02:00:00.074181,   

          gid  
0  55107008}}  
1  55107008}}  
2  55107008}}  
3  10655437}}  
4  55107008}}

In [11]:
data_short['ai5']=data_short["ai5"].str[:-1]
data_short['event']=data_short["event"].str[:-1]
data_short['timestamp']=data_short["timestamp"].str[:-1]
data_short["gid"]=data_short["gid"].str[:-2]


In [12]:
data_short

ai5    event                   timestamp  \
0       8fa683e59c02c04cb781ac689686db07  ggstart  2016-05-09 02:00:00.004906   
1       335644267c1d5f04eaea7bc6f51b1861  ggstart  2016-05-09 02:00:00.033775   
2       ca3071aad676bc963795a2b09635cdf0   ggstop  2016-05-09 02:00:00.036721   
3       17412dec7d3d02c9b0b1c3d1c3571c5c   ggstop  2016-05-09 02:00:00.041006   
4       f81167c854f1a0c86cab6188f9995824  ggstart  2016-05-09 02:00:00.074181   
5       17412dec7d3d02c9b0b1c3d1c3571c5c  ggstart  2016-05-09 02:00:00.084916   
6       a2659df45c8d05f326225fa5b1063ac9  ggstart  2016-05-09 02:00:00.093961   
7       b8bbef76f8dfee2fe190a283cd5a19a7  ggstart  2016-05-09 02:00:00.099308   
8       e8ebfc3f39512eda3aa0702b13ffed63  ggstart  2016-05-09 02:00:00.105905   
9       988e4873861347113519fbee6dd1c3b0  ggstart  2016-05-09 02:00:00.156922   
10      583361d66ad8b0827cd08d3a5d64af89   ggstop  2016-05-09 02:00:00.241238   
11      d775dbe1ca0e2610ae5505e7f489ece0  ggstart  2016-05-09 02:00:00.306023   
12      ed278fbc0dc987010e0262260cf39e16  ggstart  2016-05-09 02:00:00.380952   
13      5593edde722a92a27a8cd6c6df646a2b   ggstop  2016-05-09 02:00:00.413781   
14      96d2d9897d3e9a6da64e3db3f6143042  ggstart  2016-05-09 02:00:00.435793   
15      d23e73d5004ca552c345f6ac967febc6  ggstart  2016-05-09 02:00:00.466813   
16      a2659df45c8d05f326225fa5b1063ac9   ggstop  2016-05-09 02:00:00.486603   
17      a2659df45c8d05f326225fa5b1063ac9  ggstart  2016-05-09 02:00:00.509603   
18      ce41aa9e1b857b4ad73eda943e081981   ggstop  2016-05-09 02:00:00.536296   
19      b87ab47c55bf64f1b0aef4ca15a94784  ggstart  2016-05-09 02:00:00.553366   
20      b761fc0b92bb8b07435a05d8e835b1c0   ggstop  2016-05-09 02:00:00.578340   
21      efeb9c00bb6e1aef1b2464974b467ce2   ggstop  2016-05-09 02:00:00.666362   
22      825dcd80c73027b802ae7fd74d6a2d1d  ggstart  2016-05-09 02:00:00.668457   
23      5cf9733b0b22ac05dbb3290917c15848   ggstop  2016-05-09 02:00:00.676655   
24      df9752bca47ff5efb3dd85291eaca875   ggstop  2016-05-09 02:00:00.690341   
25      df9752bca47ff5efb3dd85291eaca875  ggstart  2016-05-09 02:00:00.694609   
26      aca6d26817f1898723ff6c7abe3ee9dd   ggstop  2016-05-09 02:00:00.697425   
27      eb963f31a491d3728c91401224dcd46a   ggstop  2016-05-09 02:00:00.713753   
28      ae7fad6cc5588799290db2ace9202cd5   ggstop  2016-05-09 02:00:00.756608   
29      8789f66052955c1ac263644755262040   ggstop  2016-05-09 02:00:00.798782   
...                                  ...      ...                         ...   
171691  85017e55381b917a659ba621bec7bee9  ggstart  2016-05-09 02:59:59.385222   
171692  77abb14b6ad93efb332b7553795d3009  ggstart  2016-05-09 02:59:59.432998   
171693  9435de7b2b3a73ea643624ed44d2abb0   ggstop  2016-05-09 02:59:59.455130   
171694  101385c65f52db5f4f6635806970634c  ggstart  2016-05-09 02:59:59.489672   
171695  5130e5d4c43fa331b50e97af5c28196d   ggstop  2016-05-09 02:59:59.533755   
171696  1b8a111a2a4893a5c6e0c7256f1e1a3d   ggstop  2016-05-09 02:59:59.582278   
171697  c1364c35aef88e5cbe8519e1b4814025   ggstop  2016-05-09 02:59:59.584391   
171698  c781133f49133e0f5a93e83048e62ed8  ggstart  2016-05-09 02:59:59.613041   
171699  327af4f56158297bf1acd0809cca0484  ggstart  2016-05-09 02:59:59.602744   
171700  7132a0aa7fa549885af73c51246c4c6d   ggstop  2016-05-09 02:59:59.616743   
171701  1b8a111a2a4893a5c6e0c7256f1e1a3d   ggstop  2016-05-09 02:59:59.657883   
171702  8927c370fde3e4c23afd5cda287cd4cd   ggstop  2016-05-09 02:59:59.671898   
171703  b3d2226b7ba071b947b0227dd6e401d5   ggstop  2016-05-09 02:59:59.678543   
171704  4acfbd34b2925238cf84abbe8e4ac71b   ggstop  2016-05-09 02:59:59.729384   
171705  4acfbd34b2925238cf84abbe8e4ac71b  ggstart  2016-05-09 02:59:59.732784   
171706  ca210a098d9361f01da944c612b7be25  ggstart  2016-05-09 02:59:59.738973   
171707  1b8a111a2a4893a5c6e0c7256f1e1a3d  ggstart  2016-05-09 02:59:59.762916   
171708  9a771f06fc7519d7fe083046bbad0214  ggstart  2016-05-09 02:59:

In [19]:
data_short.describe()

ai5    event                   timestamp  \
count                             171721   171721                      171721   
unique                             17419        2                      171694   
top     228559c219720e998b1d353daa67281f  ggstart  2016-05-09 02:00:57.064580   
freq                                 245    88786                           2   

             gid  
count     171721  
unique        20  
top     55107008  
freq      107457

# Approach-1 (final one)

In [21]:
data_gstart = data_short[data_short['event'] == 'ggstart']


data_gstop = data_short[data_short['event'] == 'ggstop']

In [22]:
data_gstart.sort_values(by=['ai5','timestamp'])
data_gstop.sort_values(by=['ai5','timestamp'])


print data_gstart.shape

print data_gstop.shape

(88786, 4)
(82935, 4)


In [23]:
data_gstart.groupby('ai5').count()

event  timestamp  gid
ai5                                                    
000523f72f16359e9962fd25c825eebf      1          1    1
00057368e3db67e8db27dc415209cc92      7          7    7
0006df67ddcf576f4d38edd46c10b5a1      1          1    1
00071e48377752c551d9ecda33c55945     13         13   13
000c6c4d3976e75a32d422bdda89f9e5      5          5    5
000d2688374c44397a3f94dd975cffb4      4          4    4
000e3141a0b69405f75c877396a988ba     10         10   10
00129e2c2eeb77cd45de41d5e8440934      3          3    3
00145d2801465edb6be8cacbc0e12052      1          1    1
0017e70ebdfdca2662c8892f811e5c66      6          6    6
0017f165b1ff5a98d293ad687dc63fa0      1          1    1
0018d7805ba3166e698953d74dec8873     41         41   41
001a771f29e534ecf5b58b39dcfe23f0     14         14   14
001c929c3c4427df51abd6d5feabca9d      1          1    1
001ed07dffa3285347a69281741cc0ba      9          9    9
00218b1750842df76166d475f717e7fa      2          2    2
0022c8c20373eebacd6909472c2db734      3          3    3
00255bee2365bf59f6dcb7fd01fcbe3b      4          4    4
00269f0bc15b02a1c619b8845df449bd     19         19   19
002b6aac41268e91705d592f5e7b1232      3          3    3
002fc6661bc602286cda7fa44f366de7     14         14   14
0034164a61d3cea9ece1c00372f58052      2          2    2
003780ead9bfb7c0d05ef120c4e7dfab      1          1    1
003c93b16d381f62d9ad3f26efeca840      8          8    8
004241c229807c81b35d2f3de005d774      4          4    4
004354bb1d644afd81247ea45a84be35      4          4    4
00486a6e9ac6528718d8d51ced33dbe0      4          4    4
0049e1daa7ff1dfb0757552680ada905      1          1    1
0049e1f49f702615bdc2245d02794774      5          5    5
005133f24025cea55fa72a4657e2fcd8     17         17   17
...                                 ...        ...  ...
ff753aa93f4c1818016516542a03a008      1          1    1
ff75d4deaba80ad1da04a2a5ec0039a7      3          3    3
ff7766e2029a6f887ab2e018f701d2c7     39         39   39
ff79b2a6ef1d91f2a5bceb72a9af6cc0      7          7    7
ff7d22c5b1efacaed6cba954de3243e0      1          1    1
ff7dd948e8a4622c83322431dd9a8d5c      1          1    1
ff90f4e719e44b4fecdc18f722669a2a      1          1    1
ff9ff72594d3b7f05589d09fbc9c0af6     14         14   14
ffa00a71721fd2c1e0c23131e2628024      1          1    1
ffa2c094280ac5b7153b4d3964d67e45      6          6    6
ffa30ffa3bdc84537f30509fdebaf06b      1          1    1
ffa3348d382cd85f090d900e36448f15     19         19   19
ffac89d29565f454e4327261d6d8861c      4          4    4
ffb703629637ae4983d86fd50db63c99      2          2    2
ffb868cf13d28d6fe8b6e140ced05d7e     61         61   61
ffbf49bdd729bcc265852fd671de683d      5          5    5
ffc01c9b14155bc74a4b6a0773894725     21         21   21
ffc447f3dfeea286bdc9759764c0dcbf      1          1    1
ffc464b1e94763d39058efa8d93f028d      2          2    2
ffc537301d473dff24e3c0d626b75970      6          6    6
ffc8c827af0be72ce8ec74cdd74479cb      7          7    7
ffce6e04a38eeef5a00131564e5f3ebf      1          1    1
ffe2bd1deca1a6644c85ffc99b0d7fa1      4          4    4
ffe2ef6f5ef5bcc7c0faba8bdb02b138      2          2    2
ffe6f19965da247e4b86caf3834f3d46      1          1    1
ffebbc1eee99ab7d04be95474e1c7883      4          4    4
ffeda1cc600c1a4fbd1c3943a2525950      2          2    2
fff4521f8bafe9f2db45ae0d78080480      1          1    1
fff897871cc2879f0ce109403e15f82f      2          2    2
fff9be87950ea0e7475cbab39d5de84f      4          4    4

[16963 rows x 3 columns]

In [25]:
data_gstop.groupby('ai5').count()

event  timestamp  gid
ai5                                                    
00057368e3db67e8db27dc415209cc92      7          7    7
0006df67ddcf576f4d38edd46c10b5a1      1          1    1
00071e48377752c551d9ecda33c55945     12         12   12
000c6c4d3976e75a32d422bdda89f9e5      5          5    5
000d2688374c44397a3f94dd975cffb4      1          1    1
000e3141a0b69405f75c877396a988ba      9          9    9
00129e2c2eeb77cd45de41d5e8440934      2          2    2
00145d2801465edb6be8cacbc0e12052      1          1    1
0017e70ebdfdca2662c8892f811e5c66      7          7    7
0017f165b1ff5a98d293ad687dc63fa0      1          1    1
0018d7805ba3166e698953d74dec8873     40         40   40
001a771f29e534ecf5b58b39dcfe23f0     10         10   10
001c929c3c4427df51abd6d5feabca9d      1          1    1
001ed07dffa3285347a69281741cc0ba      7          7    7
00218b1750842df76166d475f717e7fa      2          2    2
0022c8c20373eebacd6909472c2db734      4          4    4
00255bee2365bf59f6dcb7fd01fcbe3b      5          5    5
00269f0bc15b02a1c619b8845df449bd     16         16   16
002b6aac41268e91705d592f5e7b1232      3          3    3
002fc6661bc602286cda7fa44f366de7     13         13   13
0034164a61d3cea9ece1c00372f58052      2          2    2
003780ead9bfb7c0d05ef120c4e7dfab      1          1    1
003c93b16d381f62d9ad3f26efeca840      7          7    7
004241c229807c81b35d2f3de005d774      4          4    4
004354bb1d644afd81247ea45a84be35      4          4    4
00486a6e9ac6528718d8d51ced33dbe0      3          3    3
0049e1daa7ff1dfb0757552680ada905      1          1    1
0049e1f49f702615bdc2245d02794774      4          4    4
005133f24025cea55fa72a4657e2fcd8     16         16   16
0051c7a43a7a92d059c501ae27347a90      9          9    9
...                                 ...        ...  ...
ff753aa93f4c1818016516542a03a008      1          1    1
ff75d4deaba80ad1da04a2a5ec0039a7      3          3    3
ff7766e2029a6f887ab2e018f701d2c7     39         39   39
ff79b2a6ef1d91f2a5bceb72a9af6cc0      8          8    8
ff7d22c5b1efacaed6cba954de3243e0      2          2    2
ff7dd948e8a4622c83322431dd9a8d5c      1          1    1
ff90f4e719e44b4fecdc18f722669a2a      1          1    1
ff9ff72594d3b7f05589d09fbc9c0af6     15         15   15
ffa00a71721fd2c1e0c23131e2628024      1          1    1
ffa2c094280ac5b7153b4d3964d67e45      5          5    5
ffa30ffa3bdc84537f30509fdebaf06b      2          2    2
ffa3348d382cd85f090d900e36448f15     20         20   20
ffac89d29565f454e4327261d6d8861c      3          3    3
ffb703629637ae4983d86fd50db63c99      1          1    1
ffb868cf13d28d6fe8b6e140ced05d7e     33         33   33
ffbf49bdd729bcc265852fd671de683d      5          5    5
ffc01c9b14155bc74a4b6a0773894725     20         20   20
ffc464b1e94763d39058efa8d93f028d      2          2    2
ffc537301d473dff24e3c0d626b75970      3          3    3
ffc8c827af0be72ce8ec74cdd74479cb      5          5    5
ffce6e04a38eeef5a00131564e5f3ebf      3          3    3
ffd6b2c4e90e00c22d0e97ff314c187f      2          2    2
ffe2bd1deca1a6644c85ffc99b0d7fa1      4          4    4
ffe2ef6f5ef5bcc7c0faba8bdb02b138      2          2    2
ffe6f19965da247e4b86caf3834f3d46      1          1    1
ffebbc1eee99ab7d04be95474e1c7883      4          4    4
ffeda1cc600c1a4fbd1c3943a2525950      3          3    3
fff4521f8bafe9f2db45ae0d78080480      1          1    1
fff897871cc2879f0ce109403e15f82f      1          1    1
fff9be87950ea0e7475cbab39d5de84f      3          3    3

[15247 rows x 3 columns]

In [26]:
len(data_gstart.ai5.unique())

16963

In [27]:
len(data_gstop.ai5.unique())

15247

In [28]:
gstart_uniques_ai5 = pd.Series(data_gstart.ai5.unique())
gstop_uniques_ai5 = pd.Series(data_gstop.ai5.unique())

In [48]:
gstop_uniques_ai5.head()

0    ca3071aad676bc963795a2b09635cdf0
1    17412dec7d3d02c9b0b1c3d1c3571c5c
2    583361d66ad8b0827cd08d3a5d64af89
3    5593edde722a92a27a8cd6c6df646a2b
4    a2659df45c8d05f326225fa5b1063ac9
dtype: object

In [30]:
test = gstart_uniques_ai5.isin(gstop_uniques_ai5)

In [47]:
test.head()

0    True
1    True
2    True
3    True
4    True
dtype: bool

In [39]:
new_data=data_short


In [40]:
new_data = new_data[new_data.ai5 != 'NA']

In [46]:
new_data.head()

ai5    event                   timestamp  \
0  8fa683e59c02c04cb781ac689686db07  ggstart  2016-05-09 02:00:00.004906   
1  335644267c1d5f04eaea7bc6f51b1861  ggstart  2016-05-09 02:00:00.033775   
2  ca3071aad676bc963795a2b09635cdf0   ggstop  2016-05-09 02:00:00.036721   
3  17412dec7d3d02c9b0b1c3d1c3571c5c   ggstop  2016-05-09 02:00:00.041006   
4  f81167c854f1a0c86cab6188f9995824  ggstart  2016-05-09 02:00:00.074181   

        gid  
0  55107008  
1  55107008  
2  55107008  
3  10655437  
4  55107008

In [81]:
new_data['timestamp'] = new_data['timestamp'].astype('datetime64[ns]')

In [82]:
new_data['timestamp'] = new_data['timestamp'].astype('datetime64[ns]')

In [83]:
gstart_final_data = new_data[new_data.event == 'ggstart']

In [84]:
gstart_final_data.head()

ai5    event                  timestamp  \
32735  000523f72f16359e9962fd25c825eebf  ggstart 2016-05-09 02:13:27.016709   
3438   00057368e3db67e8db27dc415209cc92  ggstart 2016-05-09 02:01:31.466113   
8050   00057368e3db67e8db27dc415209cc92  ggstart 2016-05-09 02:03:26.785422   
17527  00057368e3db67e8db27dc415209cc92  ggstart 2016-05-09 02:07:18.759831   
23884  00057368e3db67e8db27dc415209cc92  ggstart 2016-05-09 02:09:56.364989   

            gid  
32735  55107008  
3438   55107008  
8050   55107008  
17527  55107008  
23884  55107008

In [85]:
gstop_final_data = new_data[new_data.event == 'ggstop']

In [86]:
gstop_final_data.head()

ai5   event                  timestamp  \
7995   00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:03:25.459793   
17445  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:07:16.908268   
23839  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:09:55.139737   
30766  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:12:41.991913   
32017  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:13:11.091407   

            gid  
7995   55107008  
17445  55107008  
23839  55107008  
30766  55107008  
32017  55107008

In [87]:
gstart_final_data = gstart_final_data[['ai5','event','timestamp','gid']]
gstart_final_data.reset_index(inplace=True)

In [88]:
gstart_final_data.head()

index                               ai5    event  \
0  32735  000523f72f16359e9962fd25c825eebf  ggstart   
1   3438  00057368e3db67e8db27dc415209cc92  ggstart   
2   8050  00057368e3db67e8db27dc415209cc92  ggstart   
3  17527  00057368e3db67e8db27dc415209cc92  ggstart   
4  23884  00057368e3db67e8db27dc415209cc92  ggstart   

                   timestamp       gid  
0 2016-05-09 02:13:27.016709  55107008  
1 2016-05-09 02:01:31.466113  55107008  
2 2016-05-09 02:03:26.785422  55107008  
3 2016-05-09 02:07:18.759831  55107008  
4 2016-05-09 02:09:56.364989  55107008

In [89]:
gstop_final_data = gstop_final_data[['ai5','event','timestamp','gid']]
#gstop_final_data['time'] = gstop_final_data.timestamp.astype('int64')
gstop_final_data.reset_index(inplace=True)

In [112]:
gstop_final_data.head()

index                               ai5   event                  timestamp  \
0   7995  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:03:25.459793   
1  17445  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:07:16.908268   
2  23839  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:09:55.139737   
3  30766  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:12:41.991913   
4  32017  00057368e3db67e8db27dc415209cc92  ggstop 2016-05-09 02:13:11.091407   

        gid  
0  55107008  
1  55107008  
2  55107008  
3  55107008  
4  55107008

In [92]:
final_table = pd.DataFrame()
#final_table = final_table['timeseconds']
final_table['timeseconds'] = gstop_final_data['timestamp'] - gstart_final_data['timestamp']

In [94]:
def filterSixtySeconds(t):
    delta = datetime.timedelta(seconds=60)
#     t -= delta
#     if delta > datetime.timedelta(0):
#         t += datetime.timedelta(seconds=60)

    if t > delta:
        return True

    return 0

In [97]:
final_table['valid_session'] = final_table.timeseconds.apply(filterSixtySeconds)

In [99]:
final_table.head()

timeseconds valid_session
0 -1 days +23:49:58.443084             0
1          00:05:45.442155          True
2          00:06:28.354315          True
3          00:05:23.232082          True
4          00:03:14.726418          True

In [100]:
final_table[final_table.valid_session == True].count()

timeseconds      39929
valid_session    39929
dtype: int64

In [104]:
final_table.shape

(88786, 2)

# Total Valid Sessions 
valid_session    39929

# Total Sessions
Total_session    88786

In [108]:
final_valid_table= final_table[final_table.valid_session == True]

In [110]:
final_valid_table.timeseconds.mean()

Timedelta('0 days 00:20:30.398046')

# Average session time for valid sessions

'0 days 00:20:30.398046'